### PR TITLE
Separating out link tests from service catalog dependent tests

### LINKED ISSUE #2261: Linking is never tested on OpenShift 4.x!
make test-cmd-link-unlink is never executed on OpenShift CI.

Not all test there require ServiceCatalog, we can run those on OpenShfit CI

### PR DESCRIPTION
**Pull request by prietyc123**
## What is the purpose of this change? What does it change?

Running link and unlink test on 4.2, 4.1 and 3.11 cluster
## Was the change discussed in an issue?
fixes #2261 

## How to test changes?

check 4.2, 4.1 and 3.11 test result in CI

### CODE REVIEW

File: tests/integration/servicecatalog/cmd_service_test.go [hunk 1]
```diff
@@ -279,4 +279,153 @@ var _ = Describe("odo service command tests", func() {
...
 		})
 	})
+
+	Context("When link backend between component and service", func() {
+		JustBeforeEach(func() {
+			context = helper.CreateNewContext()
+			os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+			project = helper.CreateRandProject()
+			context1 = helper.CreateNewContext()
+			context2 = helper.CreateNewContext()
```
Comment (amitkrout) [inline tests/integration/servicecatalog/cmd_service_test.go]: These lines are common in test spec that you updated form link test file. You can create a separate function that only does above operation and call the function here.

File: tests/integration/servicecatalog/cmd_service_test.go [hunk 2]
```diff
@@ -279,4 +279,153 @@ var _ = Describe("odo service command tests", func() {
...
+			originalDir = helper.Getwd()
+		})
+		JustAfterEach(func() {
+			helper.Chdir(originalDir)
+			helper.DeleteProject(project)
+			helper.DeleteDir(context)
+			helper.DeleteDir(context1)
+			helper.DeleteDir(context2)
```
Comment (amitkrout) [inline tests/integration/servicecatalog/cmd_service_test.go]: https://github.com/openshift/odo/pull/2300#discussion_r337462244

File: scripts/openshiftci-presubmit-integrationtests.sh [hunk 3]
```diff
@@ -27,6 +27,7 @@ make test-cmd-app
 make test-cmd-project
 make test-cmd-url
 make test-cmd-push
+make test-cmd-link-unlink
```
Comment (amitkrout) [inline scripts/openshiftci-presubmit-integrationtests.sh]: +1

**Comment by prietyc123** [comment]
@cdrage Can you please do next round of review

File: tests/integration/servicecatalog/cmd_service_test.go (hunk 1 above)
Comment (amitkrout) [reply tests/integration/servicecatalog/cmd_service_test.go]: @prietyc123 Use something like this 
```
setup := func() {
		project = helper.CreateRandProject()
		context1 = helper.CreateNewContext()
		context2 = helper.CreateNewContext()
	}

	removeSetup := func() {
		helper.DeleteProject(project)
		helper.DeleteDir(context)
		helper.DeleteDir(context1)
		helper.DeleteDir(context2)
	}
```
And call these in the required place

**Comment by amitkrout** [comment]
> /test v4.3-integration
> /test v4.1-integration

Wait till #2363 and #2362 go in first

**Comment by prietyc123** [comment]
> I'm assuming these files are simply just being moved from servicecatalog to the tests/integration folder? 

@cdrage Yes, and some test spec those are purely dependent on service catalog are left behind in the same service catalog file.

File: tests/integration/servicecatalog/cmd_service_test.go [hunk 4]
```diff
@@ -30,6 +30,24 @@ var _ = Describe("odo service command tests", func() {
...
+	setup := func() {
+		context = helper.CreateNewContext()
+		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+		project = helper.CreateRandProject()
+		context1 = helper.CreateNewContext()
+		context2 = helper.CreateNewContext()
+		originalDir = helper.Getwd()
+	}
+
+	removeSetup := func() {
```
Comment (amitkrout) [inline tests/integration/servicecatalog/cmd_service_test.go]: removeSetup -> cleanSetup

File: tests/integration/servicecatalog/cmd_service_test.go [hunk 5]
```diff
@@ -30,6 +30,24 @@ var _ = Describe("odo service command tests", func() {
 		oc = helper.NewOcRunner("oc")
 	})
 
+	setup := func() {
```
Comment (amitkrout) [inline tests/integration/servicecatalog/cmd_service_test.go]: setup -> preSetup

**Comment by amitkrout** [comment]
> New changes are detected. LGTM label has been removed.

It was just a rebase
/lgtm

**Comment by amitkrout** [comment]
#2299 
/test v4.3-integration

**Pull request merged**